.github/workflows/tests_extension.yml

```diff
@@ -0,0 +1,85 @@
+name: Integration tests
+
+on:
+  workflow_dispatch:
+    workflow: "*"
+  label:
+    types: [created]
+  pull_request_review:
+    types: [edited]
+
+jobs:
+  integration_test:
+    if: ${{ github.event.label.name == 'run-extension-tests' }}
+    name: Extension_${{ matrix.EXTENSION_VERSION }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        EXTENSION_VERSION: ['release', 'dev']
+
+    env:
+      MPLBACKEND: agg
+      EXTENSION: kikuchipy lumispy pyxem atomap
+      TEST_DEPS: pytest pytest-xdist pytest-rerunfailures pytest-instafail
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: conda-incubator/setup-miniconda@master
+        with:
+          miniforge-variant: Mambaforge
+          miniforge-version: latest
+          # use base environment, so that when using pip, this is from the
+          # mambaforge distribution
+          auto-activate-base: true
+          activate-environment: ""
+
+      - name: Conda info
+        run: |
+          conda info
+          conda list
+
+      - name: Install extensions and Test dependencies
+        run: |
+          mamba install hyperspy-base ${{ env.EXTENSION }} ${{ env.TEST_DEPS }}
+
+      - name: Conda list
+        run: |
+          conda list
+
+      - name: Install HyperSpy
+        run: |
+          pip install .
+
+      - name: Install Extension Dev
+        if: contains(matrix.EXTENSION_VERSION, 'dev')
+        run: |
+          pip install https://github.com/lumispy/lumispy/archive/main.zip
+          pip install https://github.com/pyxem/kikuchipy/archive/main.zip
```

Comment:
kikuchipy branch should be `develop`, which is the development branch, not `main`, since it (almost always) only contains already released stuff.